### PR TITLE
Allow order summary headings to be customized via entities

### LINKED ISSUE #52043: [Enhancement]: Make top-level order summary titles editable
### Describe the solution you'd like

In the order summary, the merchant should be able to edit the titles of each item, e.g. change "Shipping" to "Delivery".

Image

~We could explore using entities to save these so it applies across both the Cart and Checkout blocks.~ In #52892, we tried using entities but realized that the drawbacks outweigh the benefits in this case.

```[tasklist]
### Headings to make editable
- [ ] #53544
- [ ] #53548
- [ ] #53686
- [ ] "Fee" heading in cart and checkout order summary block
- [ ] "Delivery" heading in cart and checkout order summary block
```

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by samueljseay**
### Changes proposed in this Pull Request:

Closes #52043

This feature allows the editing of various headings within the cart and checkout order summary pages. Where the headings share the same purpose we save them using entities so they can be edited once and updated in both blocks.

There are some headings I've excluded:

* Taxes. Taxes by default in the editor displays taxes as line items and the way we display them differs from non line item taxes. These are displayed with their names in a list. So we can't easily edit the generic Tax heading and even if we did it might be confusing to merchants to have a discrepancy between f/e and b/e. We could enhance this somehow at some point but for now I think it's better to leave it not editable

Also note: I have stored the "Order Summary" heading in an entity even though right now it's not shared across cart and checkout, (cart uses "content" to store it's own header right now and they're different). I thought it could be useful to do this now though and enhance both to match in future because it (could) make sense to consolidate them, but i'm open to either removing that from this PR, or changing that one to be straight update/save of a regular block attribute.

### How to test the changes in this Pull Request:

For the following headings:

* "Subtotal" heading in cart and checkout order summary block
* "Total" at bottom of cart and checkout order summary block
* "Add a coupon" heading in cart and checkout order summary block
* "Fee" heading in cart and checkout order summary block
* "Delivery" heading in cart and checkout order summary block

Test:

* Edit the heading in either cart or checkout page
* Save, note that the entity editing confirmation should be shown, note that the label for the heading you're editing makes sense
* Refresh the editor, confirm that your new text shows in the place of that heading
* Go to editor for the other page (e.g. if you were editing checkout, go to editor for cart page) and ensure your new heading text also displays there in the correct place
* Go to frontend page for cart and checkout, confirm that your new text is displayed in the right heading

Example:

https://github.com/user-attachments/assets/59a5767f-e0ce-463e-abb4-65feb606bbe9

Also confirm that before editing headings that they all display correctly.

Note for testing **fee** heading you'll need to add a mock fee so that the heading displays on front-end.
Here's some code you can run to ensure that a fee is added:

```
/**
 * Add a standard $ value surcharge to all transactions in cart / checkout
 */
add_action( 'woocommerce_cart_calculate_fees','wc_add_surcharge' ); 
function wc_add_surcharge() { 
    global $woocommerce; 

    if ( is_admin() && ! defined( 'DOING_AJAX' ) ) 
        return;
    $fee = 1.00;
    $woocommerce->cart->add_fee( 'Surcharge', $fee, true, 'standard' );  
}
```

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [ ] Patch
-   [x] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [x] Enhancement - Improvement to existing functionality

#### Message 
Allow headings in cart/checkout order summary blocks to be edited

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce-blocks/assets/js/base/components/cart-checkout/totals/coupon/index.tsx [hunk 1]
```diff
@@ -76,12 +87,14 @@ export const TotalsCoupon = ( {
 		}
 	};
 
+	const PanelComponent = isEditor ? EditorPanel : Panel;
```
Comment (samueljseay) [inline plugins/woocommerce-blocks/assets/js/base/components/cart-checkout/totals/coupon/index.tsx:90]: This is a replacement for rendering a NonInteractive wrapper, instead we render a mock panel in the editor where you can still interact with a RichText placed inside it but it won't have open/close functionality.

**Review by nielslange — COMMENTED** [review]
Thanks for working on this, @samueljseay. As discussed in our call yesterday, let's pause this implementation for now, and await feedback from other @woocommerce/rubik members in p1733356953953689-slack-C8X6Q7XQU.

**Comment by senadir** [comment]
> Taxes. Taxes by default in the editor displays taxes as line items and the way we display them differs from non line item taxes. These are displayed with their names in a list. So we can't easily edit the generic Tax heading and even if we did it might be confusing to merchants to have a discrepancy between f/e and b/e. We could enhance this somehow at some point but for now I think it's better to leave it not editable

I think heading should be editable regardless, we have 2 possible outcomes here:

- It's itemized, in which case, it will show each rate alone, in the editor, we show 2, and they're not editable.
- It's not itemized, in which case, it uses a generic "tax" label, we make that editable.

**Comment by nielslange** [comment]
> I think heading should be editable regardless, we have 2 possible outcomes here:
>
>- It's itemized, in which case, it will show each rate alone, in the editor, we show 2, and they're not editable.
>- It's not itemized, in which case, it uses a generic "tax" label, we make that editable.

@senadir How can we best communicate this to merchants? I'm concerned that allowing tax labels to be editable in some situations, but not others, could create confusion for merchants.

**Comment by samueljseay** [comment]
@senadir regarding tax heading, yeah it was relatively straightforward from code perspective, I was more concerned about editing experience. How would user's find the 2 different tax styles to edit? Right now we just display one of them, so we would need a way to change state between them so the user can edit in a way that's obvious.

**Comment by samueljseay** [comment]
There has been good discussion and we've decided to not use entities for this part. I'll close this and we'll look to just use traditional block attributes here.

**Comment by senadir** [comment]
we should just follow the setting in the editor, if taxes are itemized then they can’t edit, we may show a notice in the inspector of that block and link to tax settings page, or better, allow toggling from there as an entity.

**Pull request closed without merging**